packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/document/services/JSONExternalResourceLoaderService.java

```
@@ -58,14 +59,12 @@ public boolean canHandle(InputStream inputStream, URI resourceURI, ResourceSet r
     }
 
     @Override
-    public Optional<Resource> getResource(InputStream inputStream, URI resourceURI, ResourceSet resourceSet) {
+    public Optional<Resource> getResource(InputStream inputStream, URI resourceURI, ResourceSet resourceSet, AdapterFactoryEditingDomain adapterFactoryEditingDomain) {
         Resource resource = null;
         try {
-            var jsonResource = new JSONResourceFactory().createResource(resourceURI);
-            resourceSet.getResources().add(jsonResource);
-            jsonResource.load(inputStream, new HashMap<>());
-
-            resource = jsonResource;
+            resource = new JSONResourceFactory().createResource(resourceURI);
+            resourceSet.getResources().add(resource);
+            resource.load(inputStream, new HashMap<>());
```

Ah that's what you were talking about, I created the temp variable on purpose while rebasing your PR. With your version, the set resource with a new resource and even if the load fails with an IOException, then the resource is returned instead of an empty optional.

Unless I missed something, with your code, you would never return an empty optional in case of loading error.